python/cloudtik/providers/_private/huaweicloud/utils.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from functools import lru_cache
+from typing import Any, Dict
+
+from huaweicloudsdkcore.auth.credentials import BasicCredentials
+from huaweicloudsdkcore.http.http_config import HttpConfig
+from huaweicloudsdkecs.v2 import EcsClient
+from huaweicloudsdkecs.v2.region.ecs_region import EcsRegion
+from huaweicloudsdkeip.v2 import EipClient
+from huaweicloudsdkeip.v2.region.eip_region import EipRegion
+from huaweicloudsdknat.v2 import NatClient
+from huaweicloudsdknat.v2.region.nat_region import NatRegion
+from huaweicloudsdkvpc.v2 import VpcClient
+from huaweicloudsdkvpc.v2.region.vpc_region import VpcRegion
+
+from cloudtik.core._private.constants import env_bool
+
+logger = logging.getLogger(__name__)
+
+
+@lru_cache()
+def _client_cache(ak: str, sk: str, region: str) -> Dict[str, Any]:
+    client_map = {}
+    credentials = BasicCredentials(ak, sk)
```

Comment:
Also consider other credentials than (ak, sk). CloudTik basically utilize 3 kind of credentials.
1. the basic credentials specified in the config
2. the environment variables pointing to the credential or credential file.
3. the instance credentials for VM instance (CloudTik controller runs on head node to scale workers needs instance credential)

With a proper implementation, we can share the make_XXX_client function in code which may run at different environment.

Reply:
Got